packages/frontend/src/creation-form/components/deploy.tsx

```diff
@@ -329,8 +355,53 @@ export const Deploy = ({
         void create();
     }, [devMode, onCreate, onNext, onTx, publicClient, writeAsync]);
 
+    const handleExpirationChange = useCallback(
+        (value: Date) => {
+            const converted = dateToUnixTimestamp(value);
+            setExpirationErrorText(
+                isUnixTimestampInThePast(converted)
+                    ? t("error.expiration.past")
+                    : "",
+            );
+            onStateChange((state) => ({
+                ...state,
+                expiration: converted,
+            }));
+        },
+        [onStateChange, t],
+    );
+
     return (
         <div className="flex flex-col gap-6">
+            <DateTimeInput
+                data-testid="generic-data-step-expiration-input"
```

Comment:
This should be `data-testid="deploy-step-expiration-input"`.

Reply:
Addressed in fe81a3cde236872841d5b41edbccb599b960fdd0